### ISSUE TITLE
[BUG] Incorrect/inconsistent vaccine data

### ISSUE THREAD

**Issue by stijndcl** [disease-sh/API#979]
**TLDR**: same query with different result, same dates with different result, queries were seconds apart.

**Describe the bug**
I noticed my script that uses the API gave odd results, and decided to take a look at the output (directly from your site `disease.sh/docs/`): the API doesn't return the same data consistently. Depending on what query you use, you might get different dates, and even when using the same query. Examples are below. **Screenshots are taken from the website, so this is not at all related to whatever code I used.**

Endpoint used: https://disease.sh/v3/covid-19/vaccine/coverage/countries/

**Expected behavior**
The same dates every time I send a query (on the same day), and the same data for a certain day.

**Screenshots**
_All three of these queries were only a few seconds apart._

(Query 0 - executed locally before testing on the site) - dates returned: **18th, 19th, and 20th** of January, **112k** cases for **all three dates**.

Query 1: `Country: Belgium, last 3 days` - dates returned: **17th, 18th, and 19th** of January. All 3 of these dates have the exact **same** numbers (**100.443**). The actual current day (20th) is _not_ included in this. **This is a different result than query 0 mentioned above**! Both the _dates_ and the _numbers_ are different!
![image](https://user-images.githubusercontent.com/60451863/105206559-cbcd4d80-5b46-11eb-9de5-f6ea1a323ca9.png)

Query 2: `Country: Belgium, last 1 day` - dates returned: **20th** of January, **112k** cases.
![image](https://user-images.githubusercontent.com/60451863/105206604-da1b6980-5b46-11eb-9143-19316712d699.png)

Then, I tried last 3 days again and I got:
Query 3: `Country: Belgium, last 3 days` - dates returned: **18th, 19th, and 20th** of January - all 3 have the **same** numbers (**112.293**). This is a _different result_ than the first query, but the options used were exactly the same.
![image](https://user-images.githubusercontent.com/60451863/105206626-ddaef080-5b46-11eb-8670-6ed31a152443.png)

As you can see the **first** and **third** query provided me **different dates** (20th is included in the third one), and the numbers are _always the same_ for every day. The 18th and 19th yield different results in both queries (**110k** in the first one, and **112k in the second one**). I was not coincidentally inbetween updates of the API, as **I've had the 112k result earlier today** @ query 0 (meaning that it's returned that value earlier on, and now something different again).

I am aware that endpoints might update their data every now and then, providing different data over the course of a day as new info comes in - I don't think this issue is related to that. This would only excuse the last two being the same (as "the new numbers aren't in yet" so the numbers of "today" will be the same ones as the day before). The dates themselves changed twice,  and so did the numbers (112 -> 100 -> 112), while using the same query all 3 times.

It doesn't make sense that all 3 of the dates are consistently the same, and the dates themselves vary around. Further, I've noticed that this is not related to the Belgian stats by a coincidence - I've tried some other countries randomly and they had this issue as well.

**Comment by nibble0101** [comment]
@stijndcl  I can't reproduce the problem otherwise I would have helped fix this. Have you tested this at different times and it keeps recurring because I don't see it on my end?

**Comment by stijndcl** [comment]
@nibble0101 Do you have any idea of the timestamps on which you scrape the data? It's doable to check where this came from by going off of the commit history of the source for this endpoint (which the docs say is [this](https://github.com/owid/covid-19-data/commits/master/public/data/vaccinations/country_data/Belgium.csv)). I could provide timestamps & states of my requests as well.

**Comment by nibble0101** [comment]
@stijndcl  I did ask the fellows working on the data source while working on the vaccine data PR. They update 24 hourly though  sometimes more than once a day. I believe it is scraped 24 hourly on the server too. Essentially it looks like it was last scraped on `2021-01-18` or  the source updated for `2021-01-19` as highlighted [here](https://github.com/owid/covid-19-data/blob/master/public/data/vaccinations/vaccinations.csv#L91) but since then scraping has not been done. Regarding the exact time when scraping is done on the server, perhaps @ebwinters  can help you there. I will cross check tomorrow and see what is going on. Probably I have missed something.

**Comment by ebwinters** [comment]
hey @stijndcl I'll get more time to look at this on the weekend, will update the issue then

**Comment by ebwinters** [comment]
@stijndcl I just tested this out, and whatever issue you were having seems to be fixed:
https://disease.sh/v3/covid-19/vaccine/coverage/countries/belgium
https://disease.sh/v3/covid-19/vaccine/coverage/countries/belgium?lastdays=3
